java/src/jmri/jmrit/display/layoutEditor/LayoutConnectivity.java

```diff
@@ -199,6 +199,20 @@ public PositionablePoint getAnchor() {
         return anchor;
     }
 
-    private final static Logger log = LoggerFactory.getLogger(LayoutConnectivity.class.getName());
+    @Override
+    public boolean equals(Object o) {
+        boolean result = false; // assume failure (pessimist!)
+        if ((o != null) && o instanceof LayoutConnectivity) {
+            LayoutConnectivity lc = (LayoutConnectivity) o;
+            result = toString().equals(lc.toString());
```

Comment:
This is a very expensive way to test for equality (since you are concatenating Strings together). It is better to test each individual field that determines the results of toString().